### PR TITLE
provisioner: Do not require rsync of Leap install media

### PR DESCRIPTION
**Pull request by vuntz**
(ignore the first commit, it's from https://github.com/crowbar/crowbar-core/pull/159)

If /srv/tftpboot/opensuse-42.1/x86_64/install/ exists but does not
contain the required files, we will download the minimal bits that are
needed to allow Leap installation.

It's very debatable whether this should be done in the chef cookbook or
elsewhere, and the "security check" is quite weak (just a md5sum
check)...

### CODE REVIEW

File: crowbar_framework/config/repos-cloud.yml [hunk 1]
```diff
@@ -97,3 +97,75 @@ suse-12.1:
...
+      name: "openSUSE-Leap-Update"
+      required: "mandatory"
+      features: ["os", "ha", "ceph"]
+      repomd:
+        tag: "obsrepository://build.opensuse.org/openSUSE:Leap:42.1:Update/standard"
+        key_md5: "d8039df72c407bc7dd7c1d9c66d58627"
+      url: "http://download.opensuse.org/update/leap/42.1/oss/"
+      smt_path:
+      ask_on_error: false
+    chef-10:
```
Comment (mjura) [inline crowbar_framework/config/repos-cloud.yml]: We can use this repos

```
    crowbar-3_0:
      name: "Crowbar-3.0"
      required: "mandatory"
      repomd:
        tag: "obsrepository://build.opensuse.org/systemsmanagement:crowbar:3.0/openSUSE_Leap_42.1"
        key_md5: "081f86b11a9fc2d1a4ab0b32ec3e3812"
      url: "http://download.opensuse.org/repositories/systemsmanagement:/crowbar:/3.0/openSUSE_Leap_42.1/"
      smt_path:
      priority: 90
      ask_on_error: false
    crowbar-shared:
      name: "Crowbar-Shared"
      required: "mandatory"
      repomd:
        tag: "obsrepository://build.opensuse.org/systemsmanagement:crowbar:shared/openSUSE_Leap_42.1"
        key_md5: "081f86b11a9fc2d1a4ab0b32ec3e3812"
      url: "http://download.opensuse.org/repositories/systemsmanagement:/crowbar:/shared/openSUSE_Leap_42.1/"
      smt_path:
      priority: 90
      ask_on_error: false
```

Comment (vuntz) [inline crowbar_framework/config/repos-cloud.yml]: I explicitly didn't want these as I wanted the minimal set of packages for the nodes, to not impact anything else on Leap. I understand these are needed for the admin server, but the non-admin nodes don't need all the changed packages.

**Comment by dirkmueller** [comment]
do not merge before M8 is out.

File: chef/cookbooks/provisioner/recipes/setup_base_images.rb [hunk 2]
```diff
@@ -543,14 +546,47 @@
...
+        if missing_files && !param_install_url.nil? && !param_install_url.empty?
+          kernel_path = Pathname.new(os_dir).join("install").join(kernel)
+          kernel_tmp_path = Pathname.new(os_dir).join("install").join("#{kernel}.check")
+          initrd_path = Pathname.new(os_dir).join("install").join(initrd)
+          initrd_tmp_path = Pathname.new(os_dir).join("install").join("#{initrd}.check")
+
+          unless kernel_path.exist?
+            kernel_path.dirname.mkpath
+            `wget --timeout 5 -O #{kernel_tmp_path} #{param_install_url}/#{kernel}`
+            if Digest::MD5.file(kernel_tmp_path.to_s).hexdigest == "fcc6dc33eac95bdf96c6cea4dd3eb7c2"
```
Comment (aplanas) [inline chef/cookbooks/provisioner/recipes/setup_base_images.rb]: Is there a way to show an error in the md5 is different?

**Comment by vuntz** [comment]
Let's close for now, no interest in pushing this further.

**Pull request closed without merging**